src/main/java/at/ac/tuwien/kr/alpha/grounder/atoms/HiddenAtom.java

```diff
@@ -0,0 +1,49 @@
+package at.ac.tuwien.kr.alpha.grounder.atoms;
+
+import at.ac.tuwien.kr.alpha.common.predicates.Predicate;
+import at.ac.tuwien.kr.alpha.common.atoms.Atom;
+import at.ac.tuwien.kr.alpha.common.atoms.BasicAtom;
+import at.ac.tuwien.kr.alpha.common.terms.Term;
+import at.ac.tuwien.kr.alpha.grounder.Substitution;
+
+import java.util.List;
+
+/**
+ * An internal ordinary atom, i.e., a BasicAtom that is internal and not reported in answer sets.
+ * Copyright (c) 2017, the Alpha Team.
+ */
+public class HiddenAtom extends BasicAtom {
```

Comment:
I dislike the addition of `InternalAtom`:

 * Looking at the class, there is practically no behavior it adds to `BasicAtom` (calls to `super` everywhere).
 * We already have `BasicAtom.isInternal` which should "hide" an atom.

Proposed solution: Change the implementation of `BasicAtom.isInternal` so that it does not always return `false`, but some flag `private final boolean internal`. This flag should be `false` by default. Add a new constructor that allows to set it to `true` and use this instead of `HiddenAtom`.

Reply:
True, the `HiddenAtom` is like an ordinary atom that does not show up in answer sets.
By the way, why is the hiding not property of a `Predicate` but of an atom?

Reply:
I am not sure. But it would probably be better if we drop `Atom.isInternal` and instead replace `atom.isInternal()` by something like `isInternal(atom.getPredicate())` in `NaiveGrounder.assignmentToAnswerSet`, which then checks for some predefined instances of `Predicate`, just like in the `NaiveGrounder` constructor which calls `adaptWorkingMemoryForPredicate` for a similar set of predicates.